tests/Tests/400__ods_dashboard/400__ods_dashboard.robot

```diff
@@ -274,3 +295,20 @@ Filter By Using More Than One Filter And Check Output
     FOR    ${id}    IN    @{LIST_OF_IDS_FOR_COMBINATIONS}
         Deselect Checkbox Using Id    id=${id}
     END
+
+Verify Notification Saying Notebook Builds Not Started
+    [Documentation]     Verifies RHODS Notification Drawer Contains Names of Image Builds which have not started
+    Sleep    10s  reason=Wait For Notifications
+    Reload Page
+    RHODS Notification Drawer Should Contain    message=These notebook image builds have not started:
+    FOR    ${image}    IN    @{images}
+        RHODS Notification Drawer Should Contain    message=${image}
+    END
+
+Wait Until Remaining Builds Are Complete And Close Browser
+    [Documentation]     Waits Until Remaining builds have Status as Complete and Closes Browser
+    ${pytorch_build}=  Search Last Build    namespace=redhat-ods-applications    build_name_includes=pytorch
+    ${tensorflow_build}=  Search Last Build    namespace=redhat-ods-applications    build_name_includes=tensorflow
+    Wait Until Build Status Is    namespace=redhat-ods-applications    build_name=${pytorch_build}  expected_status=Complete
+    Wait Until Build Status Is    namespace=redhat-ods-applications    build_name=${tensorflow_build}  expected_status=Complete
+    Dashboard Test Teardown
```

Comment:
I have some doubts about if it is necessary to check if the builds failed. And if they did, then restart them. We could discuss during automation meeting

Reply:
The builds are started automatically(by cuda-thoth after its completion) , So Is there any chance that they would fail?

Reply:
Added A Check for Failing Builds

Reply:
yes, but it works only if the build fails only once. If after the restart it fails again I think this code will not catch the failure. However, it might be enough for now..what do you think @jgarciao @lugi0 @tarukumar @pablofelix @mwaykole ? I recall we had a discussion on this topic some automation meetings ago

Reply:
Seems to be addressed now with the new Test Teardown rebuilding all failed builds